code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm

```diff
@@ -0,0 +1,613 @@
+	var/static/heretic_level_generated = FALSE
+	/// A weakref to the mind of our heretic.
+	var/datum/mind/heretic_mind
+	/// Lazylist of minds that we won't pick as targets.
+	var/list/datum/mind/target_blacklist
+	/// An assoc list of [uif] to [timers] - a list of all the timers of people in the shadow realm currently
+	var/list/return_timers
+	/// Evil organs we can put in people
+	var/static/list/grantable_organs = list(
+		/obj/item/organ/internal/appendix/corrupt,
+		/obj/item/organ/internal/eyes/corrupt,
+		/obj/item/organ/internal/heart/corrupt,
+		/obj/item/organ/internal/liver/corrupt,
+		/obj/item/organ/internal/lungs/corrupt,
+	)
+
+/datum/heretic_knowledge/hunt_and_sacrifice/Destroy(force)
+	heretic_mind = null
+	LAZYCLEARLIST(target_blacklist)
+	return ..()
+
+/datum/heretic_knowledge/hunt_and_sacrifice/on_research(mob/user, datum/antagonist/heretic/our_heretic)
+	. = ..()
+	addtimer(CALLBACK(src, PROC_REF(generate_targets), user, our_heretic), 10 SECONDS)
+
+/datum/heretic_knowledge/hunt_and_sacrifice/proc/generate_targets(mob/user, datum/antagonist/heretic/our_heretic)
+	obtain_targets(user, silent = TRUE, heretic_datum = our_heretic)
+	heretic_mind = our_heretic.owner
+
+/datum/heretic_knowledge/hunt_and_sacrifice/recipe_snowflake_check(mob/living/user, list/atoms, list/selected_atoms, turf/loc)
+	var/datum/antagonist/heretic/heretic_datum = IS_HERETIC(user)
+	// First we have to check if the heretic has a Living Heart.
+	// You may wonder why we don't straight up prevent them from invoking the ritual if they don't have one -
+	// Hunt and sacrifice should always be invokable for clarity's sake, even if it'll fail immediately.
+	if(heretic_datum.has_living_heart() != HERETIC_HAS_LIVING_HEART)
+		to_chat(user, "<span class='hierophant'>The ritual failed, you have no living heart!</span>")
+		return FALSE
+
+	// We've got no targets set, let's try to set some.
+	// If we recently failed to acquire targets, we will be unable to acquire any.
+	if(!LAZYLEN(heretic_datum.sac_targets))
+		atoms += user
+		return TRUE
+
+	// If we have targets, we can check to see if we can do a sacrifice
+	// Let's remove any humans in our atoms list that aren't a sac target
+	for(var/mob/living/carbon/human/sacrifice in atoms)
+		// If the mob's not in soft crit or worse, remove from list
+		if(sacrifice.health > 0)
+			atoms -= sacrifice
+		// Otherwise if it's neither a target nor a cultist, remove it
+		else if(!(sacrifice in heretic_datum.sac_targets) && !IS_CULTIST(sacrifice))
+			atoms -= sacrifice
+
+	// Finally, return TRUE if we have a target in the list
+	if(locate(/mob/living/carbon/human) in atoms)
+		return TRUE
+
+	// or FALSE if we don't
+	to_chat(user, "<span class='hierophant'>The ritual failed, no valid sacrifice was found!</span>")
+	return FALSE
+
+/datum/heretic_knowledge/hunt_and_sacrifice/on_finished_recipe(mob/living/user, list/selected_atoms, turf/loc)
```

Comment:
Please change this variable name to something that isn't `loc`, that's a really nasty thing to shadow

Reply:
oh shit yeah, will have to fix that

Reply:
replaced, thanks!